### PR TITLE
Added code changes to pom-versioned.xml to support osgi bundling for kryo5.

### PR DESCRIPTION
**Pull request by Shradha-git**

### CODE REVIEW

**Comment by theigl** [comment]
@Shradha-git: Thank you!

Is it really necessary to specify all sub-packages like this:

```
com.esotericsoftware.kryo.kryo${kryo.major.version}.objenesis.instantiator,
com.esotericsoftware.kryo.kryo${kryo.major.version}.objenesis.strategy,
```

Is this not enough?

`com.esotericsoftware.kryo.kryo${kryo.major.version}.*`

File: pom-versioned.xml [hunk 1]
```diff
@@ -82,6 +82,33 @@
...
+						<Import-Package><![CDATA[
+						COM.jrockit*;resolution:=optional,
+						COM.newmonics*;resolution:=optional,
+						jrockit*;resolution:=optional,
+						sun.reflect;resolution:=optional,
+						sun.misc;resolution:=optional,
+						sun.nio.ch;resolution:=optional
+						]]>
+						</Import-Package>
+						<Export-Package>com.esotericsoftware.kryo.kryo${kryo.major.version},com.esotericsoftware.kryo.kryo${kryo.major.version}.asm,com.esotericsoftware.kryo.kryo${kryo.major.version}.io,
```
Comment (theigl) [inline pom-versioned.xml:100]: If we really need all the separate packages listed here, please add a newline after each package to make this easier to read.

**Review by theigl — CHANGES_REQUESTED** [review]

File: pom-versioned.xml (hunk 1 above)
Comment (Shradha-git) [reply pom-versioned.xml:100]: @theigl  Initially I had tried adding the same like com.esotericsoftware.kryo.kryo${kryo.major.version}.*
But this change didn't work and will not create any export package entry in MANIFEST.MF file
Reason : There is no source java files in this versioned project, and hence it doesn't find anything to export when using the "*" config. So we need to manually list out each package.
cc @vikinghawk

Comment (theigl) [reply pom-versioned.xml]: @Shradha-git: I tried to find a better solution for this but the shade and bundle plugin don't play well together:

https://stackoverflow.com/questions/18280608/using-maven-bundle-plugin-with-the-maven-shade-plugin.

So it seems we really have to list all export packages, but I think we might be able to get rid of some of the exports. The `Export-Package` definition for the un-shaded JAR only lists the following two packages in addition to Kryo's own packages:

```
org.objenesis.instantiator,
org.objenesis.strategy
```

All other packages do not need to be exposed to consumers of the bundle. Could you please try this simplified export configuration and check if it works:

```
<Export-Package>
    com.esotericsoftware.kryo.kryo${kryo.major.version},
    com.esotericsoftware.kryo.kryo${kryo.major.version}.io,			
    com.esotericsoftware.kryo.kryo${kryo.major.version}.serializers,
    com.esotericsoftware.kryo.kryo${kryo.major.version}.unsafe,
    com.esotericsoftware.kryo.kryo${kryo.major.version}.util,		
    com.esotericsoftware.kryo.kryo${kryo.major.version}.objenesis,
    com.esotericsoftware.kryo.kryo${kryo.major.version}.objenesis.instantiator,
    com.esotericsoftware.kryo.kryo${kryo.major.version}.objenesis.strategy		
</Export-Package>
```

Comment (Shradha-git) [reply pom-versioned.xml]: @theigl  Tried with above simplified list of Export-Package and it is getting resolved. I will update the Export-Package list with above  one.

**MANIFEST.MF generated:**
Manifest-Version: 1.0
Automatic-Module-Name: com.esotericsoftware.kryo.kryo5
Bnd-LastModified: 1617716489841
Build-Jdk: 10.0.2
Built-By: SR068348
Bundle-Description: Fast, efficient Java serialization. This is the ve
 rsion specific Kryo artifact.
Bundle-License: https://opensource.org/licenses/BSD-3-Clause
Bundle-ManifestVersion: 2
Bundle-Name: Kryo 5
Bundle-SymbolicName: com.esotericsoftware.kryo.5
Bundle-Version: 5.0.4
Created-By: Apache Maven Bundle Plugin
Export-Package: com.esotericsoftware.kryo.kryo5;version="5.0.4",com.es
 otericsoftware.kryo.kryo5.io;version="5.0.4",com.esotericsoftware.kry
 o.kryo5.serializers;version="5.0.4",com.esotericsoftware.kryo.kryo5.u
 nsafe;version="5.0.4",com.esotericsoftware.kryo.kryo5.util;version="5
 .0.4",com.esotericsoftware.kryo.kryo5.objenesis;version="5.0.4",com.e
 sotericsoftware.kryo.kryo5.objenesis.instantiator;version="5.0.4",com
 .esotericsoftware.kryo.kryo5.objenesis.strategy;version="5.0.4"
Import-Package: sun.misc;resolution:=optional,sun.nio.ch;resolution:=o
 ptional,sun.reflect;resolution:=optional
Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))"
Tool: Bnd-4.2.0.201903051501

**Review by theigl — APPROVED** [review]

**Comment by theigl** [comment]
Thanks @Shradha-git! There are still some warnings by the shade and bundle plugins. I'll try to get rid of them after merging.

**Pull request merged**